app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,7 @@
+class RecipesController < ApplicationController
+  def show 
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredients
+    @recipe_cost = @recipe.recipe_cost
```

Comment:
This is fine, but you can also access the `ingredients` association in the view without any MVC concerns, as well as the `recipe_cost` method. This would also help keep your controllers "skinny".